commit bff5867ccf45ffd4d3974629cfb3581739b5c4e1 (refs/changes/10/1210/1)
Author: David Pacheco <dap@joyent.com>
Date:   2017-01-05T11:10:37-08:00 (2 years, 9 months ago)
    
    joyent/manta-mlive#3 document command-line options
    joyent/manta-mlive#4 allow target shard count to be configured on the command line
    joyent/manta-mlive#5 move to cr.joyent.us and update contribution guidelines

diff --git a/CONTRIBUTING.md b/CONTRIBUTING.md
new file mode 100644
index 0000000..208d435
--- /dev/null
+++ b/CONTRIBUTING.md
@@ -0,0 +1,5 @@
+# Contributing
+
+This repository is part of the Joyent Manta project.  See the [contribution
+guidelines for the Manta
+project](https://github.com/joyent/manta/blob/master/CONTRIBUTING.md).
diff --git a/bin/mlive b/bin/mlive
index 74ed75e..959d6dd 100755
--- a/bin/mlive
+++ b/bin/mlive
@@ -8,7 +8,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -72,7 +72,7 @@ function main()
 	};
 
 	parser = new mod_getopt.BasicParser('f(force)p:(path)S(skip-setup)' +
-	    '9(help)', process.argv);
+	    's:(shards)9(help)', process.argv);
 	while ((option = parser.getopt()) !== undefined) {
 		switch (option.option) {
 		case 'f':
@@ -93,6 +93,14 @@ function main()
 			args.ml_paths.push(option.optarg);
 			break;
 
+		case 's':
+			args.ml_nshards = parseInt(option.optarg, 10);
+			if (isNaN(args.ml_nshards) || args.ml_nshards <= 0) {
+				usage('unsupported number of shards: "%s"',
+				    args.ml_nshards);
+			}
+			break;
+
 		default:
 			mod_assertplus.equal('?', option.option);
 			usage();
@@ -138,12 +146,20 @@ function usage()
 	}
 
 	console.error([
-	    sprintf('usage: %s [-f | --force] [-p | --path PATH] REPORT_TIME',
+	    sprintf('usage: %s [-f | --force] [-p | --path PATH] [-S] ' +
+	        '[-s | --shards NSHARDS] REPORT_INTERVAL',
 	        mlArg0),
 	    '',
 	    'Runs a small, continuous read and write load against a Manta ',
 	    'service in order to assess liveness.  Prints out when outages ',
-	    'appear to start and end on a per-second interval.'
+	    'appear to start and end, as well as a general report every ',
+	    'REPORT_INTERVAL seconds.',
+	    '',
+	    '    -f, --force       clobber content in target paths',
+	    '    -p, --path PATH   specify Manta paths to write test objects',
+	    '    -S, --skip-setup  skip initialization (must already be done)',
+	    '    -s, --shards N    create enough directories to test ',
+	    '                      N metadata shards'
 	].join('\n'));
 	process.exit(2);
 }
diff --git a/package.json b/package.json
index ddc8d26..3974f8a 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
 	"name": "mlive",
-	"version": "0.0.1",
+	"version": "1.0.0",
 	"description": "Manta liveness tester",
 	"bin": {
 		"mlive": "bin/mlive"
@@ -10,17 +10,17 @@
 	      	"url": "https://github.com/joyent/manta-mlive"
 	},
 	"keywords": [ "manta", "liveness" ],
-	"author": "Dave Pacheco <dap@joyent.com>",
+	"author": "David Pacheco <dap@joyent.com>",
 	"license": "MPL-2.0",
 	"bugs": { "url": "https://github.com/joyent/manta-mlive/issues" },
 	"homepage": "https://github.com/joyent/manta-mlive",
 	"dependencies": {
-		"assert-plus": "0.1.5",
-		"bunyan": "1.5.1",
-		"extsprintf": "1.3.0",
-		"manta": "2.0.3",
-		"posix-getopt": "1.2.0",
-		"vasync": "1.6.3",
-		"verror": "1.6.0"
+		"assert-plus": "^1.0.0",
+		"bunyan": "^1.5.1",
+		"extsprintf": "^1.3.0",
+		"manta": "^4.1.1",
+		"posix-getopt": "^1.2.0",
+		"vasync": "^1.6.3",
+		"verror": "^1.6.0"
 	}
 }
